src/python/pants/backend/awslambda/python/target_types.py

```diff
@@ -2,36 +2,107 @@
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
 import re
+from dataclasses import dataclass
 from typing import Match, Optional, Tuple, cast
 
+from pants.backend.python.dependency_inference.module_mapper import PythonModule, PythonModuleOwners
+from pants.backend.python.dependency_inference.rules import import_rules
 from pants.backend.python.target_types import InterpreterConstraintsField, PythonSources
 from pants.core.goals.package import OutputPathField
 from pants.engine.addresses import Address
+from pants.engine.rules import Get, collect_rules, rule
 from pants.engine.target import (
     COMMON_TARGET_FIELDS,
+    AsyncFieldMixin,
     Dependencies,
+    InjectDependenciesRequest,
+    InjectedDependencies,
     InvalidFieldException,
     StringField,
     Target,
+    WrappedTarget,
 )
+from pants.engine.unions import UnionRule
+from pants.option.subsystem import Subsystem
 
 
-class PythonAwsLambdaSources(PythonSources):
-    expected_num_files = range(0, 1)
+class PythonAwsLambdaDefaults(Subsystem):
+    """Default settings for the `python_awslambda` target."""
 
+    options_scope = "python-awslambda-defaults"
```

Comment:
Open to suggestions on what to name this. This isn't technically a default...it's more an option related to `python_awslambda` targets.

We can't use `python-awslambda` because that's confusing between `./pants help python_awslambda` (target) vs. `./pants help python-awslambda` (subsystem).

Reply:
I really like that the help system overhaul pulled these two concepts closer together, because it makes that kind of potential confusion more obvious. But this is an interesting case... will let Benjy comment.

Reply:
Yeah, these aren't really defaults for the python_awslambda target type, so this name is a touch misleading.

How about 

`python-awslambda-infer` 

To match `python-infer`, 

and the option can just be `--handler`, again matching the convention in `python-infer`.

Reply:
`pex-binary-defaults` is probably not the right place for inference-related options. The name, and docstring, imply that this is for defaults that can be overridden on a per-target basis. I.e., each option corresponds to a field on the corresponding target type. And that is not the case for these options (or might it be in the future?)

Reply:
It probably isn't the right place. We could put the Python binaries one on `python-infer`, `[python-infer].pex_binary_entry_point`. 

But then we lose symmetry with `[python-awslambda_infer]handler`. Perhaps it should be `[pex-binary-infer].entry_point`.

My only concern is the proliferation of subsystems. Will we have `[shiv-binary-infer]` and `[pyinstaller-binary-infer]`? 

--

In a perfect world, we would simply put the AWS Lambda option on `[python-infer]`. We only can't do this because there is no way to add on an option to a pre-existing subsystem.

Reply:
Now that I think of it, do we actually need an option here at all? When would we not want to infer these? How about just letting them be controlled by the global python inference option?

Reply:
You might not want it to avoid the perf cost of creating a module mapping. Imo, we should always have a mechanism to completely disable dep inference.

> How about just letting them be controlled by the global python inference option?

We could do that via `[python-infer].imports`. It is a bit of a misnomer, though. This isn't really import analysis - it's solely looking at the module name in a BUILD file, which is why I thought we should add two new options.

Reply:
Maybe we should always use this dep then. It's in a build file, as you say, so there is no "inference". It's more like an explicit dep. When might someone want to turn this off?

Reply:
Well, assuming we want an option to gate this on, we can create a general "--entry-points" option on python-infer and reuse that for pex_binary, python_awslambda and in the future any other python binary-ish target types.